stamina-core/src/main/scala/stamina/Persister.scala

```diff
@@ -18,8 +19,17 @@ abstract class Persister[T: ClassTag, V <: Version: VersionInfo](val key: String
   def canUnpersist(p: Persisted): Boolean = p.key == key && p.version <= currentVersion
 
   private[stamina] def convertToT(any: AnyRef): Option[T] = any match {
-    case t: T ⇒ Some(t)
-    case _    ⇒ None
+    case t: T ⇒ t match {
+      case tagged: TypeTagged[_] if typeTagOption.isDefined ⇒
+        val typeTag = tagged.tag
+        val currentTypeTag = typeTagOption.get.tpe.toString
+        typeTag.tpe.toString match {
```

Comment:
You're comparing the String represenations of the type tags here. 

Would comparing with `=:=` be too strict for some cases? (http://docs.scala-lang.org/overviews/reflection/symbols-trees-types.html)